client/src/main/python/slipstream/SlipStreamHttpClient.py

```diff
@@ -288,6 +285,15 @@ def get_server_configuration(self):
         _, config = self._retrieve(self.configuration_endpoint)
         return config
 
+    def login(self, username, password):
```

Comment:
Not sure it will.  There should still be way to handshake to the server's /auth/login to get a generic token.
API keys will allow handshake only to the resources they were issued for.  Thus, it will be either `authorization: Basic <api-key>` or some other custom header when contacting the specific resource.